### PR TITLE
Refactoring cxg utility classes in preparation for CXG conversion tooling

### PR DESCRIPTION
**Pull request by maniarathi**
This PR is the precursor to the PR that will actually move the CXG tool over to the CLI. My thought-process was the the transformation from H5AD to CXG will need to be modularized in the future so that it can be used for the eventual automated pipeline. Accordingly in this PR, I moved as much of the type conversion, sparsity checks, other utility functions out into the `utils` directory and wrote thorough unit tests. 

You can see the rest of the work in the draft PR here: https://github.com/chanzuckerberg/cellxgene/pull/1701.

### CODE REVIEW

File: server/common/utils/matrix_utils.py [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+
+        column_shift[start_column_index:end_column_index] = matrix_subset_mode.mode
+        number_of_non_zero_elements += total_number_of_rows * (end_column_index - start_column_index) - np.sum(
+            matrix_subset_mode.count
+        )
+
+        if number_of_non_zero_elements > maximum_number_of_non_zero_elements_in_matrix:
+            if end_column_index != total_number_of_columns:
+                logging.info(
+                    "Matrix is not sparse even with column shift. Percentage of non-zero elements (estimate): %6.2f"
```
Comment (MDunitz) [inline server/common/utils/matrix_utils.py:100]: nit -- I think you can use f strings and still get the formatting options.

File: server/common/utils/sanitization_utils.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
+import re
+
+
+def sanitize_values_in_list(list_of_keys: list):
+    """
+    Returns a dictionary mapping of the old keys in the list of `list_of_keys` to its new, clean name that is both
+    safe and unique.
+    """
+
+    if not [all(isinstance(key, str) for key in list_of_keys)]:
```
[the remark this answers is not in the document]
Comment (maniarathi) [reply server/common/utils/sanitization_utils.py]: Oops, the all was in the wrong place. Fixed and fixed the tests that should have caught this.

File: server/common/utils/sanitization_utils.py [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+        raise Exception("List of keys to sanitize must contain all strings.")
+
+    # Mask out [~/.] and anything outside the ASCII range.
+    mask = re.compile(r"[^ -\-0-\[\]-\}]")
+    clean_keys_list = [mask.sub("_", key) for key in list_of_keys]
```
[the remark this answers is not in the document]
Comment (mckinsel) [reply server/common/utils/sanitization_utils.py:15]: Oh yeah you're right. I guess I thought `sub` was replacing the first match only, but that is extremely not true.

File: server/common/utils/matrix_utils.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+                    "Matrix is not sparse. Percentage of non-zero elements (exact): %6.2f"
+                    % (100 * number_of_non_zero_elements / total_number_of_matrix_elements)
+                )
+            return False
+
+    is_sparse = (100.0 * number_of_non_zero_elements / total_number_of_matrix_elements) < sparse_threshold
+    return is_sparse
+
+
+def is_matrix_sparse_with_column_shift_encoding(matrix, sparse_threshold):
```
[the remark this answers is not in the document]
Comment (mckinsel) [reply server/common/utils/matrix_utils.py]: Ummmm no good suggestions....maybe just `get_column_shift_encode_for_matrix` and returning `None` suggests there isn't one? Totally fine to just leave it.

**Pull request merged**